MUXSDKKaltura/MUXSDKKaltura/MUXSDKPlayerBinding.swift

```diff
@@ -347,12 +347,13 @@ public class MUXSDKPlayerBinding: NSObject {
         let videoDataUpdated = videoData.hasUpdates || liveUpdates || renditionUpdates
 
         if self.videoData.sourceDimensionsHaveChanged, self.videoData.size.equalTo(self.videoData.lastDispatchedVideoSize) {
-            let sourceDimensions = player.sourceDimensions
-            if !self.videoData.size.equalTo(sourceDimensions) {
-                self.videoData.size = sourceDimensions
-                
-                if sourceDimensions.width > 0, sourceDimensions.height > 0 {
-                    self.videoData.sourceDimensionsHaveChanged = false
+            if let sourceDimensions = player.currentItem?.sourceDimensions {
```

Comment:
oh interesting, were we not getting correct dimensions previously?

Reply:
hehe no, it's the same just that the way of getting the currentItem is very specific to each player while the dimensions logic is just depending on the AvPlayerItem itself, so I changed it on the shared components to be a property in an extension of avPlayerItem. So here I had to extract the item first and then call the sourceDimensions computer property on the item itself.